py/embedded-server/java-runtime/src/main/java/io/deephaven/python/server/EmbeddedServer.java

```diff
@@ -0,0 +1,105 @@
+import org.jpy.PyObject;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+public class EmbeddedServer {
+    @Singleton
+    @Component(modules = {
+            DeephavenApiServerModule.class,
+            HealthCheckModule.class,
+            PythonPluginsRegistration.Module.class,
+            JettyServerModule.class
+    })
+    public interface PythonServerComponent extends DeephavenApiServerComponent {
+        @Component.Builder
+        interface Builder extends DeephavenApiServerComponent.Builder<PythonServerComponent.Builder> {
+            PythonServerComponent build();
+        }
+
+        void injectFields(EmbeddedServer instance);
+    }
+
+    @Inject
+    DeephavenApiServer server;
+    @Inject
+    Scheduler scheduler;
+    @Inject
+    Provider<ScriptSession> scriptSession;
+
+    public EmbeddedServer(int port, PyObject dict) throws IOException {
+        final Configuration config = Main.init(new String[0], EmbeddedServer.class);
+        PyDictWrapper pyConfig = dict.asDict();
+
+        int httpSessionExpireMs = config.getIntegerWithDefault("http.session.durationMs", 300000);
+        int schedulerPoolSize = config.getIntegerWithDefault("scheduler.poolSize", 4);
+        int maxInboundMessageSize = config.getIntegerWithDefault("grpc.maxInboundMessageSize", 100 * 1024 * 1024);
+
+        DaggerEmbeddedServer_PythonServerComponent
+                .builder()
+                .withPort(port)
+                .withSchedulerPoolSize(schedulerPoolSize)
+                .withSessionTokenExpireTmMs(httpSessionExpireMs)
+                .withMaxInboundMessageSize(maxInboundMessageSize)
+                .withOut(null)
+                .withErr(null)
+                .build()
+                .injectFields(this);
+    }
+
+    public void start() throws Exception {
+        server.run();
+        System.out.println("Server started on port " + server.server().getPort());
+        new Thread(() -> {
+            try {
+                checkGlobals(scriptSession.get(), null);
+                server.join();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }).start();
+    }
+
+    private void checkGlobals(ScriptSession scriptSession, @Nullable ScriptSession.SnapshotScope lastSnapshot) {
+        ScriptSession.SnapshotScope nextSnapshot;
+        try {
+            nextSnapshot = scriptSession.snapshot(lastSnapshot);
+        } catch (IllegalStateException e) {
+            if (e.getMessage().startsWith("Expected transition from=")) {
+                // We are limited in how we can track external changes, and the web IDE has made this change and
+                // already applied it.
+                // Take a fresh snapshot right away to continue polling
+                nextSnapshot = scriptSession.snapshot();
```

Comment:
If you run a command via ConsoleService.ExecuteCommand(), it will apply its own diff. Then, when this thread runs, it will also notice the change, and the validation in the appmode wiring (...even if app mode is disabled?) will get grumpy at the duplicate add or no-op modify - this prevents that exception from killing the thread (and in turn, the whole jvm, and then the process).